api/organization.go

```diff
@@ -0,0 +1,150 @@
+package api
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"fmt"
+	"github.com/banzaicloud/banzai-types/components"
+	"github.com/banzaicloud/pipeline/auth"
+	"github.com/banzaicloud/pipeline/model"
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+)
+
+//OrganizationMiddleware parses the organization id from the request, queries it from the database and saves it to the current context
+func OrganizationMiddleware(c *gin.Context) {
+	log := logger.WithFields(logrus.Fields{"tag": "OrganizationMiddleware"})
+	orgidParam := c.Param("orgid")
+	orgid, err := strconv.ParseUint(orgidParam, 10, 32)
+	if err != nil {
+		message := fmt.Sprintf("error parsing organization id: %q", orgidParam)
+		log.Info(message)
+		c.AbortWithStatusJSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: message,
+			Error:   message,
+		})
+	}
+
+	user := auth.GetCurrentUser(c.Request)
+	var organization = auth.Organization{ID: uint(orgid)}
+	var organizations []auth.Organization
+
+	db := model.GetDB()
+	err = db.Model(user).Where(&organization).Related(&organizations, "Organizations").Error
+	if err != nil {
+		message := "error fetching organizations"
+		log.Info(message + ": " + err.Error())
+		c.AbortWithStatusJSON(http.StatusInternalServerError, components.ErrorResponse{
+			Code:    http.StatusInternalServerError,
+			Message: message,
+			Error:   message,
+		})
+		return
+	}
+
+	if len(organizations) != 1 {
+		message := fmt.Sprintf("organization not found: %q", orgidParam)
```

Comment:
If we need to check for `len(organizations) > 1` as well than this error message is missleading.